lib/cli/src/Cardano/CLI.hs

```diff
@@ -296,6 +331,202 @@ runCli = join . customExecParser preferences
   where
     preferences = prefs showHelpOnEmpty
 
+{-------------------------------------------------------------------------------
+                            Commands - 'HD Derivation'
+-------------------------------------------------------------------------------}
+
+cmdKey :: Mod CommandFields (IO ())
+cmdKey = command "key" $ info (helper <*> cmds) $ mempty
+    <> progDesc "Derive keys from mnemonics."
+  where
+    cmds = subparser $ mempty
+        <> cmdRootKey
+
+-- | Parse a key type
+--
+-- Note that we in the future might replace the type @ByronWalletStyle@ with
+-- another type, to include Shelley keys.
+parseKeyType :: Parser (IO ByronWalletStyle)
+parseKeyType = toIO . fromText <$> strOption
+    ( long "type"
+    <> metavar "KEYTYPE"
+    <> helpDoc (Just (vsep typeOptions))
+    )
+  where
+    toIO :: Either TextDecodingError ByronWalletStyle -> IO ByronWalletStyle
+    toIO (Right x) =
+        return x
+    toIO (Left (TextDecodingError e)) =
+        hPutStrLn stderr e >> exitFailure
+
+    typeOptions = string <$>
+        [ "Any of the following:"
+        , "  random (" ++ allowedWords Random ++ ")"
+        , "  icarus (" ++ allowedWords Icarus ++ ")"
+        , "  trezor (" ++ allowedWords Trezor ++ ")"
+        , "  ledger (" ++ allowedWords Ledger ++ ")"
+        ]
+
+    allowedWords = (++ " words")
```

Comment:
I think this is something we should have near the `AllowedWords` type family declaration.
What about:

```hs
prettyAllowedWords
    :: forall style.
        ( NatVals (AllowedMnemonics style)
        )
    => Proxy style
    -> String
prettyAllowedWords _ =
    show $ natVals $ Proxy @(AllowedMnemonics style) -- TODO: pretty formatting of english enumeration
```

Reply:
Eh, maybe in a separate pr please. We should maybe have a notion of `all` wallet styles also. And might be best to have at the term-level.